src/charm.py

```diff
@@ -361,6 +393,7 @@ def _on_synapse_nginx_pebble_ready(self, _: ops.HookEvent) -> None:
             return
         logger.debug("synapse_nginx_pebble_ready replanning nginx")
         pebble.replan_nginx(container)
+        self._generate_nginx_config()
```

Comment:
Two things - are the lines not in the proper order or I'm missing something - shouldn't it be _generate_nginx_config first and then replan_nginx? Second is aren't these functions better placed in pebble.py?

Reply:
Moved!